### PR TITLE
feat: add ability to specify custom bus environments

### PR DESCRIPTION
**Pull request by jrcrawfo**
## Why?
* [JIRA](): To use this bus service as a dependency, we need to support custom environments on initialization of the class so we can hold those specific configurations outside of this repo

## What changed?
- [x] Added an additional parameter to specify custom bus environments
- [x] Updated the channel stats to be compatible with Pulsar 1.7+

### CODE REVIEW

**Review by kkikendall — APPROVED** [review]

File: contxt/services/bus.py [hunk 1]
```diff
@@ -21,7 +21,9 @@ class MessageBusService(ConfiguredApi):
         ),
     )
 
-    def __init__(self, auth: Auth, organization_id: str, env: str = "production", **kwargs) -> None:
+    def __init__(self, auth: Auth, organization_id: str, custom_envs: List[ApiEnvironment] = None, env: str = "production", **kwargs) -> None:
+        if custom_envs:
+            MessageBusService._envs = custom_envs
```
Comment (branchvincent) [inline contxt/services/bus.py:26]: modifying a static class variable in an instance's initializer feels off, can consumers just set `MessageBusService._envs` once directly? that was the intention, ie:

```python
# Customize your environment
MessageBusService._envs += (...)

service = MessageBuseService()
```

**Comment by branchvincent** [comment]
linter failures can be reproduced with:
```
$ poetry run pre-commit run -a
```

File: contxt/services/bus.py (hunk 1 above)
Comment (jrcrawfo) [reply contxt/services/bus.py]: good call -- subclass made way more sense. thanks!

Comment (branchvincent) [reply contxt/services/bus.py]: awesome

**Review by branchvincent — APPROVED** [review]

**Review by mgagliardo91 — APPROVED** [review]

**Pull request merged**